### PR TITLE
[rancher-logging] Update chart for new PSP approach

### PR DESCRIPTION
**Pull request by stormqueen1990**
### Changes
* Include validation to check if the PSP API is present in the target cluster.
* Standardize the PSP switch.
* Bump the chart version to 102.0.0.

### Testing

- [x] Ensure a `helm template <release-name> <chart-path>` never fails
- [x] Ensure a `helm template --set global.cattle.psp.enabled=true <release-name> <chart-path>` never fails

#### Kubernetes < v1.25 specific scenarios

- [x] Ensure a `helm install --dry-run <release-name> <chart-path>` does not fail when your `KUBECONFIG` is pointing to a 1.24 cluster
- [x] Ensure a `helm install --dry-run --set global.cattle.psp.enabled=true <release-name> <chart-path>` does not fail when your `KUBECONFIG` is pointing to a 1.24 cluster
- [x] Ensure a `helm install <release-name> <chart-path>` does not fail when your `KUBECONFIG` is pointing to a 1.24 cluster
- [x] Ensure a `helm install --set global.cattle.psp.enabled=true <release-name> <chart-path>` does not fail when your `KUBECONFIG` is pointing to a 1.24 cluster

#### Kubernetes >= v1.25 specific scenarios

- [x] Ensure a `helm install --dry-run <release-name> <chart-path>` does not fail when your `KUBECONFIG` is pointing to a 1.25 cluster
- [x] Ensure a `helm install --dry-run --set global.cattle.psp.enabled=true <release-name> <chart-path>` does fail when your `KUBECONFIG` is pointing to a 1.25 cluster
- [x] Ensure a `helm install <release-name> <chart-path>` does not fail when your `KUBECONFIG` is pointing to a 1.25 cluster
- [x] Ensure a `helm install --set global.cattle.psp.enabled=true <release-name> <chart-path>` does fail when your `KUBECONFIG` is pointing to a 1.25 cluster, with the error from `validate-psp-install.yaml` showing up

### CODE REVIEW

File: packages/rancher-logging/package.yaml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 url: https://kubernetes-charts.banzaicloud.com/charts/logging-operator-3.17.10.tgz
-version: 101.1.0
+version: 102.0.0
```
Comment (pennyscissors) [inline packages/rancher-logging/package.yaml:2]: 102 would be for a Rancher 2.8 version line (100 is 2.6, 101 is 2.7, and so on). I think this should be 101.2.0 if the change merits a minor version bump

Comment (pennyscissors) [reply packages/rancher-logging/package.yaml:2]: Discussed offline, lgtm

**Review by prachidamle — APPROVED** [review]
LGTM

File: release.yaml [hunk 2]
```diff
@@ -52,8 +52,10 @@ rancher-istio:
 - 101.1.0+up1.15.3
 rancher-logging:
 - 101.1.0+up3.17.10
```
Comment (eliyamlevy) [inline release.yaml:54]: For some reason your PR has the 101 versions of the chart still in the release yaml. You might have missed [this pr](https://github.com/rancher/charts/pull/2390/files#diff-a0b29149f10adefb5c44814777ca3aa3c992d81fd92a1a051add31fab99452fcL56) in the rebase?

Comment (stormqueen1990) [reply release.yaml:54]: The removal of the unreleased version happened after this PR was merged.

Comment (eliyamlevy) [reply release.yaml:54]: oops ignore, wrong pr

**Pull request merged**